browser/js/common/factories/mixboard-factory.js

```diff
@@ -82,5 +82,10 @@ app.factory('MixBoardFactory', function(){
             }
         }
     };
+
+    MixBoardFactory.saveSegment = function(track){
+        track.start = track.region.start;
```

Comment:
I'd put a check in here to see if a region exists. If one doesn't, we shouldn't change the start/end values